### PR TITLE
chore: change api key resource's value property

### PR DESCRIPTION
**Pull request by aaythapa**
### Description of changes

The value property had underscores in it which caused some integ tests to fail. 

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [x] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by GavinZZ** [comment]
Is this a new test? If not , why does it only fail now?

**Review by ConnorRobertson — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]

**Comment by aaythapa** [comment]
Holding off on merging for a bit, got some errors while running the integ test loally

**Comment by aaythapa** [comment]
> Is this a new test? If not , why does it only fail now?

Old test but failing now because of potentially breaking change from another team, changing here to unblock us

**Pull request merged**
